### DISCUSSION TITLE
Working rich with pyats

### DISCUSSION THREAD

**Discussion by opnine** [Textualize/rich#1153] [Q&A]
Hi all,

Im working with pyats to get the output from my networking devices, and the idea is to add columns and tables to this output, I wrote the script but can make it work, I would like to know if someone had already done  this and can send me any examples.

thanks.

**Comment by willmcgugan** [comment]
There's an [example](https://rich.readthedocs.io/en/latest/tables.html) in the docs. What problemsare you having?

**Reply by opnine** [reply]
Hello sir.
The issue is I cant  get the pyats output to be printed inside the rich tables/columns, when I execute the script, it I can see the print with rich tables/columns,  but not with the output I want, and I know this is my bad for missing something inside the code. 
thanks

**Reply by willmcgugan** [reply]
If you can post some code I may be able to offer some suggestions...

**Reply by opnine** [reply]
Hello sir,, 
Here is my code, I applogize if is not clear im new in python and I try to learn with examples.

```
#All my python modules
import time
from rich.console import Console
from rich.table import Table
from genie.testbed import load

#Test bed file and connect to the device defined.
testbed = load('testbed.yaml')
device = testbed.devices['SW_53']
print("Connecting...")
device.connect(log_stdout=False)

#Build out the table using Rich.table  j
table = Table(title="Show CDP Neighbors")
table.add_column("Device_id", style="red", no_wrap=True)
table.add_column("Local Interface", style="white")
table.add_column("Hold Time")
table.add_column("Port ID")

console = Console()

table.add_row(preoutput)

#for k, v in preoutput['cdp'].items():
for k, v in preoutput['cdp']["index"].items():

    deid = v.get('device_id')
    loint = v.get('local_interface')
    holdt = v.get('hold_time')
    portid = v.get('port_id')


#    if deid ==  " ":
#         table.add_row(k, deid)
#    elif holdt >= 0:
#         table.add_row(k , holdt)
```
This is the output I what to add to the columns/rows.

```
{
  "cdp": {
    "index": {
      "1": {
        "capability": "R S I",
        "device_id": "SW22.lab.com",
        "hold_time": 150,
        "local_interface": "Ethernet0/0",
        "platform": "",
        "port_id": "LinuxUniEth0/0"
      },

```

**Reply by willmcgugan** [reply]
Any error messages?

I see this line but no assignment to `preoutput`. 

```python
table.add_row(preoutput)
```

When you call `add_row` you will typically need one item per column. So it might be something like `table.add_row(deid, loint, holdt, portid)`.

If you are just learning Python you might want to try posting on https://www.reddit.com/r/learnpython/

**Reply by opnine** [reply]
Thanks for the tips, I will keep working on it!

**Reply by opnine** [reply]
I made it print in the firts column with you examples. thanks!!

**Comment by opnine** [comment]
Thanks for all the tips, I managed to print 2 of the 4 columns,

thanks.

On Fri, Apr 2, 2021 at 8:41 AM Will McGugan ***@***.***>
wrote:

> Any error messages?
>
> I see this line but no assignment to preoutput.
>
> table.add_row(preoutput)
>
> When you call add_row you will typically need one item per column. So it
> might be something like table.add_row(deid, loint, holdt, portid).
>
> If you are just learning Python you might want to try posting on
> https://www.reddit.com/r/learnpython/
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/willmcgugan/rich/discussions/1153#discussioncomment-562100>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AEZUVVZ7X7K27PUWKIX4K63TGWUNLANCNFSM42IU6T4A>
> .
>


-- 

Juraj Anton Papic
Tel +54 911 3445 6944
Skype juraj.papic
